pkgs/by-name/kc/kcc/package.nix

```
@@ -0,0 +1,62 @@
+{
+  stdenv,
+  lib,
+  qt6,
+  python3Packages,
+  fetchFromGitHub,
+  p7zip,
+  archiveSupport ? true,
+}:
+python3Packages.buildPythonApplication rec {
+  pname = "kcc";
+  version = "6.2.2";
+
+  src = fetchFromGitHub {
+    owner = "ciromattia";
+    repo = "kcc";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-GX0WPu1EZx/8H1ajivxgnym4jECmwPBJU25vzcfOQCw=";
+  };
+
+  nativeBuildInputs = [ qt6.wrapQtAppsHook ];
+
+  buildInputs = lib.optional stdenv.hostPlatform.isLinux qt6.qtwayland;
+  propagatedBuildInputs = (
+    with python3Packages;
+    [
+      packaging
+      pillow
+      psutil
+      python-slugify
+      raven
+      requests
+      natsort
+      mozjpeg_lossless_optimization
+      distro
+      pyside6
+    ]
+  );
+
+  qtWrapperArgs = lib.optionals archiveSupport [
+    "--prefix"
+    "PATH"
+    ":"
+    "${lib.makeBinPath [ p7zip ]}"
+  ];
+
+  postFixup = ''
+    wrapProgram $out/bin/kcc "''${qtWrapperArgs[@]}"
```

I think it'll be necessary to wrap at least `kcc-c2e` to have the bin path of `p7zip` available. I'm guessing it isn't necessary for the `kcc-c2p` script.

```suggestion
    wrapProgram $out/bin/kcc "''${qtWrapperArgs[@]}"
    wrapProgram $out/bin/kcc-c2e --prefix PATH : ${lib.makeBinPath [ p7zip ]}
```

Also, is `qtWrapperArgs` necessary to wrap `kcc` with explicitly or is it used by the wrapper automatically?

---

I guess I didn't take the archive option into account in my suggestion there